File: cli/src/main/java/com/gmail/marcosav2010/command/base/InfoCMD.java
package com.gmail.marcosav2010.command.base;

import com.gmail.marcosav2010.command.Command;
import com.gmail.marcosav2010.main.Main;

class InfoCMD extends Command {

    InfoCMD() {
        super("info", new String[]{"i"}, "[peer] (empty = manager)");
    }

    @Override
    public void execute(String[] arg, int args) {
        if (args == 0) {
            Main.getInstance().getPeerManager().printInfo();

        } else {
            String target = arg[0];
            if (Main.getInstance().getPeerManager().exists(target)) {
                Main.getInstance().getPeerManager().get(target).printInfo();
            }
        }
    }
}

File: cli/src/main/java/com/gmail/marcosav2010/command/base/InfoCMD.kt
package com.gmail.marcosav2010.command.base

import com.gmail.marcosav2010.command.Command
import com.gmail.marcosav2010.main.Main

internal class InfoCMD : Command("info", arrayOf("i"), "[peer] (empty = manager)") {

    override fun execute(arg: Array<String>, length: Int) {
        val peerManager = Main.instance.peerManager!!

        if (length == 0) {
            peerManager.printInfo()
        } else {
            val target = arg[0]
            if (peerManager.exists(target))
                peerManager[target]!!.printInfo()
        }
    }
}